src/formatDate.js

```diff
@@ -50,7 +50,29 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateArray = date.split(fromFormat[3]);
+  const dateTransform = {};
+  const dateToFormat = [];
+
+  for (let i = 0; i < dateArray.length; i++) {
+    dateTransform[fromFormat[i]] = dateArray[i];
+  }
+
+  if (toFormat[toFormat.indexOf('YYYY')] && dateTransform.YY) {
+    dateTransform.YYYY = (dateTransform.YY < 30)
```

Comment:
its hard to understand what's happening here, use more vars
set proper keys proper names for object keys